### PR TITLE
[BUGFIX] Move unnecessary dependencies to devDependencies

### PR DESCRIPTION
**Pull request by Gaurav0**
For https://github.com/emberjs/data/issues/6255

Created a new package, -test-infra, that should be able to be in devdependencies only.

Also hoists and syncs all devdependencies and dependencies.

### CODE REVIEW

**Comment by rwjblue** [comment]
> I noticed that -build-infra only contains node js files, so made it a devDependency wherever used.

I think we actually do need `-build-infra` as a dependency in each of the packages. Specifically, they are used by the other packages in the `index.js` for building the addon itself (which is definitely needed when consumed).

**Comment by runspired** [comment]
Confirm what Rob said. The test-helper bits should probably be in a different package from the build-utils bits. I'm thinking `@ember-data/-build-infra` for these built utils and moving the stuff that can be in dev-deps to a new `@ember-data/-dev-infra` package.

File: packages/adapter/node-tests/blueprints/adapter-test.js [hunk 1]
```diff
@@ -12,10 +12,10 @@ const expect = chai.expect;
 
 const SilentError = require('silent-error');
 
-const generateFakePackageManifest = require('@ember-data/-build-infra/src/node-test-helpers/generate-fake-package-manifest');
-const fixture = require('@ember-data/-build-infra/src/node-test-helpers/fixture');
+const generateFakePackageManifest = require('../../../-test-infra/src/node-test-helpers/generate-fake-package-manifest');
```
Comment (runspired) [inline packages/adapter/node-tests/blueprints/adapter-test.js]: relative path should be absolute

File: packages/adapter/node-tests/blueprints/adapter-test.js [hunk 2]
```diff
@@ -12,10 +12,10 @@ const expect = chai.expect;
...
+const fixture = require('@ember-data/-test-infra/src/node-test-helpers/fixture');
 
-const setupTestEnvironment = require('@ember-data/-build-infra/src/node-test-helpers/setup-test-environment');
+const setupTestEnvironment = require('../../../-test-infra/src/node-test-helpers/setup-test-environment');
```
Comment (runspired) [inline packages/adapter/node-tests/blueprints/adapter-test.js]: relative path should be absolute

File: packages/model/node-tests/blueprints/model-test.js [hunk 3]
```diff
@@ -9,10 +9,10 @@ const modifyPackages = blueprintHelpers.modifyPackages;
 const chai = require('ember-cli-blueprint-test-helpers/chai');
 const expect = chai.expect;
 
-const generateFakePackageManifest = require('@ember-data/-build-infra/src/node-test-helpers/generate-fake-package-manifest');
-const fixture = require('@ember-data/-build-infra/src/node-test-helpers/fixture');
+const generateFakePackageManifest = require('../../../-test-infra/src/node-test-helpers/generate-fake-package-manifest');
```
Comment (runspired) [inline packages/model/node-tests/blueprints/model-test.js]: relative path should be absolute

File: packages/model/node-tests/blueprints/model-test.js [hunk 4]
```diff
@@ -9,10 +9,10 @@ const modifyPackages = blueprintHelpers.modifyPackages;
...
+const fixture = require('@ember-data/-test-infra/src/node-test-helpers/fixture');
 
-const setupTestEnvironment = require('@ember-data/-build-infra/src/node-test-helpers/setup-test-environment');
+const setupTestEnvironment = require('../../../-test-infra/src/node-test-helpers/setup-test-environment');
```
Comment (runspired) [inline packages/model/node-tests/blueprints/model-test.js]: relative path should be absolute

File: packages/serializer/node-tests/blueprints/serializer-test.js [hunk 5]
```diff
@@ -12,10 +12,10 @@ const expect = chai.expect;
 
 const SilentError = require('silent-error');
 
-const generateFakePackageManifest = require('@ember-data/-build-infra/src/node-test-helpers/generate-fake-package-manifest');
-const fixture = require('@ember-data/-build-infra/src/node-test-helpers/fixture');
+const generateFakePackageManifest = require('../../../-test-infra/src/node-test-helpers/generate-fake-package-manifest');
```
Comment (runspired) [inline packages/serializer/node-tests/blueprints/serializer-test.js]: relative path should be absolute

File: packages/serializer/node-tests/blueprints/serializer-test.js [hunk 6]
```diff
@@ -12,10 +12,10 @@ const expect = chai.expect;
...
+const fixture = require('@ember-data/-test-infra/src/node-test-helpers/fixture');
 
-const setupTestEnvironment = require('@ember-data/-build-infra/src/node-test-helpers/setup-test-environment');
+const setupTestEnvironment = require('../../../-test-infra/src/node-test-helpers/setup-test-environment');
```
Comment (runspired) [inline packages/serializer/node-tests/blueprints/serializer-test.js]: relative path should be absolute

File: packages/serializer/node-tests/blueprints/transform-test.js [hunk 7]
```diff
@@ -9,10 +9,10 @@ const modifyPackages = blueprintHelpers.modifyPackages;
 const chai = require('ember-cli-blueprint-test-helpers/chai');
 const expect = chai.expect;
 
-const generateFakePackageManifest = require('@ember-data/-build-infra/src/node-test-helpers/generate-fake-package-manifest');
-const fixture = require('@ember-data/-build-infra/src/node-test-helpers/fixture');
+const generateFakePackageManifest = require('../../../-test-infra/src/node-test-helpers/generate-fake-package-manifest');
```
Comment (runspired) [inline packages/serializer/node-tests/blueprints/transform-test.js]: relative path should be absolute

File: packages/serializer/node-tests/blueprints/transform-test.js [hunk 8]
```diff
@@ -9,10 +9,10 @@ const modifyPackages = blueprintHelpers.modifyPackages;
...
+const fixture = require('@ember-data/-test-infra/src/node-test-helpers/fixture');
 
-const setupTestEnvironment = require('@ember-data/-build-infra/src/node-test-helpers/setup-test-environment');
+const setupTestEnvironment = require('../../../-test-infra/src/node-test-helpers/setup-test-environment');
```
Comment (runspired) [inline packages/serializer/node-tests/blueprints/transform-test.js]: relative path should be absolute

File: package.json [hunk 9]
```diff
@@ -36,6 +36,8 @@
   "devDependencies": {
     "@babel/plugin-transform-typescript": "^7.5.5",
     "@ember/optional-features": "^0.7.0",
+    "@ember/ordered-set": "^2.0.3",
+    "@glimmer/env": "^0.1.7",
```
Comment (runspired) [inline package.json]: glimer/env and ordered-set being added here seems suspect

File: packages/-test-infra/package.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  "license": "MIT",
+  "author": "",
+  "directories": {
+    "doc": "doc",
+    "test": "tests"
+  },
+  "scripts": {
+  },
+  "dependencies": {
+    "ember-cli-blueprint-test-helpers": "^0.19.1"
```
Comment (runspired) [inline packages/-test-infra/package.json]: I suspect more should be here (like mocha)

File: package.json [hunk 11]
```diff
@@ -45,59 +47,86 @@
     "@types/qunit": "^2.5.3",
     "@types/rsvp": "^4.0.3",
     "babel-eslint": "^10.0.2",
+    "babel-plugin-debug-macros": "^0.3.2",
```
Comment (runspired) [inline package.json]: a lot more things were added to this file than I would have thought, I feel many likely should have been placed into the test-infra package?

File: packages/-test-infra/package.json (hunk 10 above)
Comment (Gaurav0) [reply packages/-test-infra/package.json]: mocha isn't 1`require`d by any of the files.

File: package.json (hunk 9 above)
Comment (Gaurav0) [reply package.json]: They are being required in one of the packages, and I hoisted all of the dependencies to the main package.json. See comment below.

File: package.json (hunk 11 above)
Comment (Gaurav0) [reply package.json]: I hoisted all dependencies to the main package.json file, and synchronized versions. Putting them here as well as in the individual package.json files in each package ensures that the same version is used across all packages.

The other reason I did this is I got warnings running https://github.com/mybigday/yarn-workspace-sync-deps

Comment (runspired) [reply package.json]: there were a few things that needed fixed (I noticed a few dangling dev-deps with "*") but there's already a `sync` command within the repo that does what we want. It looks like this is hoisting `dependencies` in addition to `devDependencies` which is not something I think we want at this point.

Comment (Gaurav0) [reply package.json]: I got errors running that script.

**Pull request merged**